src/hooks/useFetchTheme.ts

```diff
@@ -1,34 +1,10 @@
-import { useEffect, useState } from "react";
-import { useNavigate } from "react-router-dom";
-import axios from "axios";
+import { useQuery } from "@tanstack/react-query";
 import { getThemeInfo } from "@/services/theme";
-import { showErrorToast } from "@/styles/toast";
-import type { ThemeInfo } from "@/types/theme";
 
 export function useFetchTheme(themeId?: string) {
-  const [theme, setTheme] = useState<ThemeInfo | null>(null);
-  const [loading, setLoading] = useState(true);
-  const navigate = useNavigate();
-
-  useEffect(() => {
-    const fetchTheme = async () => {
-      setLoading(true);
-      try {
-        if (!themeId) return;
-        const data = await getThemeInfo(themeId);
-        setTheme(data);
-      } catch (error) {
-        if (axios.isAxiosError(error) && error.response?.status === 404) {
-          showErrorToast("존재하지 않는 테마입니다.");
-          navigate("/", { replace: true });
-        }
-      } finally {
-        setLoading(false);
-      }
-    };
-
-    fetchTheme();
-  }, [themeId, navigate]);
-
-  return { theme, loading };
+  return useQuery({
+    queryKey: ["themeInfo", themeId],
+    queryFn: () => getThemeInfo(themeId!),
+    enabled: !!themeId,
+  });
```

Comment:
혹시 여기는 `useFetch`를 사용하지 않은 이유가 있을까요?

`useFetch`를 사용하도록 통일하고, `useFetch`를 더 범용적으로 사용할 수 있도록 옵션을 `rest paramter` 같은걸로 뚫어주면 좋을 것 같습니다!

```suggestion
  return useFetch(`/themes/${themeId}/info`, null, { enabled : !!themeId });
```